contracts/interchain-token-service/src/contract.rs

```diff
@@ -239,16 +252,49 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
     }
 
     fn deploy_remote_interchain_token(
-        _env: &Env,
-        _caller: Address,
-        _salt: BytesN<32>,
-        _minter: Option<Bytes>,
-        _destination_chain: String,
-        _gas_token: Token,
+        env: &Env,
+        caller: Address,
+        salt: BytesN<32>,
+        destination_chain: String,
+        gas_token: Token,
     ) -> Result<BytesN<32>, ContractError> {
-        // TODO: implementation
+        caller.require_auth();
+
+        let deploy_salt = Self::interchain_token_deploy_salt(env, caller.clone(), salt);
+        let token_id = Self::interchain_token_id(env, Address::zero(env), deploy_salt);
+        let registered_token_address = Self::token_address(env, token_id.clone());
```

Comment:
since the token address might not exist, query the token config manually, and return an explicit error if it doesn't exist?